lib/matplotlib/text.py

```diff
@@ -1158,6 +1159,10 @@ def set_rotation(self, s):
             The rotation angle in degrees in mathematically positive direction
             (counterclockwise). 'horizontal' equals 0, 'vertical' equals 90.
         """
+        if isinstance(s, str):
+            _api.check_in_list(['vertical', 'horizontal'], rotation=s)
+        elif s is not None:
+            cbook._check_isinstance(numbers.Real, s=s)
```

Comment:
either both should be `s=s`, or both should be `rotation=s`, depending on whether you want to strictly match the kwarg name, or use a semantically more meaningful name.

Reply:
Both should be `rotation=s`. It's understandable in the context of `set_rotation()` as well as in `Text(..., rotation=)`.
For the latter `s=s` would result in "'invalid string' is not a valid value for s" which is quite unhelpful.